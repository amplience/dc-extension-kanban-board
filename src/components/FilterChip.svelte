<script lang="ts">
  import DeleteIcon from '../assets/icons/ic-delete.svg';
  import Icon from './Icon.svelte';
  import { createEventDispatcher } from 'svelte';
  export let label: string;
  export let active: boolean = false;
  export let removeable: boolean = false;
  export let clickable: boolean = false;

  $: chipClass = [
    active ? 'active' : undefined,
    clickable ? 'clickable' : undefined,
    removeable ? 'removeable' : undefined,
  ]
    .filter(() => !undefined)
    .join(' ');

  const dispatch = createEventDispatcher();

  const onChipClick = () => dispatch('click');
  const onCloseClick = (event: Event) => {
    event.cancelBubble = true;
    dispatch('close');
  };
</script>

<style>
  div {
    min-width: 0;
    display: flex;
    background-color: hsl(0, 0%, 95%);
    padding: 6px 8px;
    border-radius: 12px;
    margin-right: 12px;
  }

  span.label {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    font-size: 13px;
    font-weight: 400;
  }

  div.active {
    background-color: #039be5;
    color: #fff;
  }

  div.clickable {
    cursor: pointer;
  }

  div.removeable > span.label {
    padding: 0 6px 0 0;
  }

  div.clickable:hover {
    background-color: #badff9;
  }
  span.cross {
    cursor: pointer;
    padding: 0;
  }

  span.cross :global(div) {
    position: relative;
  }
</style>

<div data-testid="chip" on:click={onChipClick} class={chipClass}>
  <span class="label"> {label} </span>
  {#if removeable}
    <span
      data-testid="remove-chip-button"
      class="cross"
      on:click={onCloseClick}>
      <Icon icon={DeleteIcon} width="8px" height="8px" />
    </span>
  {/if}
</div>
