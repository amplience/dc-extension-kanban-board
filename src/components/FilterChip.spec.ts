import { render } from '@testing-library/svelte';
import FilterChip from './FilterChip.svelte';

describe('FilterChip', () => {
  it('should render a Chip', () => {
    const { container } = render(FilterChip, {
      label: 'test-label',
      onClick: () => {
        /* noop */
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a clickable Chip', () => {
    const { container } = render(FilterChip, {
      label: 'test-label',
      clickable: true,
      onClick: () => {
        /* noop */
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render an active Chip', () => {
    const { container } = render(FilterChip, {
      label: 'test-label',
      active: true,
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render an clickable active Chip', () => {
    const { container } = render(FilterChip, {
      label: 'test-label',
      active: true,
      clickable: true,
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
