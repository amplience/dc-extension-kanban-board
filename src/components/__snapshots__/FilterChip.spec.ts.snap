// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterChip should render a Chip 1`] = `
<div>
  <div
    class="   svelte-h28tab"
    data-testid="chip"
  >
    <span
      class="label svelte-h28tab"
    >
      test-label
    </span>
     
  </div>
</div>
`;

exports[`FilterChip should render a clickable Chip 1`] = `
<div>
  <div
    class=" clickable  svelte-h28tab"
    data-testid="chip"
  >
    <span
      class="label svelte-h28tab"
    >
      test-label
    </span>
     
  </div>
</div>
`;

exports[`FilterChip should render an active Chip 1`] = `
<div>
  <div
    class="active   svelte-h28tab"
    data-testid="chip"
  >
    <span
      class="label svelte-h28tab"
    >
      test-label
    </span>
     
  </div>
</div>
`;

exports[`FilterChip should render an clickable active Chip 1`] = `
<div>
  <div
    class="active clickable  svelte-h28tab"
    data-testid="chip"
  >
    <span
      class="label svelte-h28tab"
    >
      test-label
    </span>
     
  </div>
</div>
`;
